SourceWriters/BitmapEffectWriter.cs

```diff
@@ -0,0 +1,188 @@
+﻿using PaintDotNet.Effects.Gpu;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace PaintDotNet.Effects
+{
+    internal static class BitmapEffectWriter
+    {
+        internal const string UsingStatements = ""
+            + "using System;\r\n"
+            + "using System.IO;\r\n"
+            + "using System.Linq;\r\n"
+            + "using System.Diagnostics;\r\n"
+            + "using System.Drawing;\r\n"
```

Comment:
I recommend not having a `using` for `System.Drawing.*` for BitmapEffects or GpuEffects. Instead, prefer using the primitives in `PaintDotNet.Rendering`, and the imaging stuff in `PaintDotNet.Imaging`. Text rendering through `PaintDotNet.DirectWrite`. I might need some more examples of how to do text rendering and bitmap manipulation, so be sure to let me know what's needed. Also, having usings for `System.Drawing.*` will cause an ambiguity when `IDeviceContext` comes into play, whether for `GpuEffect`s, or for `BitmapEffects` that want to use D2D/DWrite.

You still often need `System.Drawing.Bitmap` just for the image you pass down from the constructor. So I usually add a using alias like `using GdipBitmap = System.Drawing.Bitmap;` "Gdip" is a weird term that nobody else will understand (Gdip for "GDI+"), so maybe `DrawingBitmap` or `SystemDrawingBitmap` or `ClassicBitmap` or ... I dunno :)